File: backend/Codebymister.Domain/Entities/Outreach.cs
using Codebymister.Domain.Common;
using Codebymister.Domain.Enums;

namespace Codebymister.Domain.Entities;

public class Outreach : BaseEntity
{
    public Guid LeadId { get; private set; }
    public OutreachChannel Channel { get; private set; }
    public string Message { get; private set; }
    public DateTime SentAt { get; private set; }
    public bool Responded { get; private set; }
    public DateTime? ResponseAt { get; private set; }
    public ResponseStatus ResponseStatus { get; private set; }
    public bool FollowUpSent { get; private set; }
    public string? Notes { get; private set; }

    public Lead Lead { get; private set; }

    private Outreach() { }

    public Outreach(
        Guid leadId,
        OutreachChannel channel,
        string message,
        string? notes = null)
    {
        LeadId = leadId;
        Channel = channel;
        Message = message;
        SentAt = DateTime.UtcNow;
        Responded = false;
        ResponseStatus = ResponseStatus.NoResponse;
        FollowUpSent = false;
        Notes = notes;
    }

    public void MarkAsResponded(DateTime? responseAt, ResponseStatus status)
    {
        Responded = true;
        ResponseAt = responseAt ?? DateTime.UtcNow;
        ResponseStatus = status;
    }

    public void MarkFollowUpSent()
    {
        FollowUpSent = true;
    }

    public void UpdateNotes(string notes)
    {
        Notes = notes;
    }
}

File: dashboard/src/types/outreach.ts
export enum OutreachChannel {
  WhatsApp = 1,
  Instagram = 2,
  Email = 3,
  Phone = 4,
  LinkedIn = 5,
  Facebook = 6,
  InPerson = 7
}

export enum ResponseStatus {
  NoResponse = 1,
  Positive = 2,
  Neutral = 3,
  Negative = 4,
  NotInterested = 5
}

export interface Outreach {
  id: string;
  leadId: string;
  leadName: string;
  channel: OutreachChannel;
  message: string;
  sentAt: string;
  responded: boolean;
  responseAt?: string | null;
  responseStatus: ResponseStatus;
  followUpSent: boolean;
  notes?: string | null;
}

export interface CreateOutreachRequest {
  leadId: string;
  channel: OutreachChannel;
  message: string;
  notes?: string;
}

export interface UpdateOutreachRequest {
  responseAt?: string;
  responseStatus: ResponseStatus;
  followUpSent: boolean;
  notes?: string;
}
